File: web/WebFirstLab/WithAjax/php/Hit.php
<?php
    require_once('Containable.php');
    require_once('Resultable.php');

    class Hit implements Resultable {
        private $x;
        private $y;
        private $r;

        public function __construct(Containable $x, Containable $y, Containable $r) {
            $this->x = $x;
            $this->y = $y;
            $this->r = $r;
        }

        public function result() {
            try {
                if($this->x->content() > 0) {
                    return $this->y->content() > 0 ? $this->firstQuarter() : $this->fourthQuarter();
                } else { 
                    return $this->y->content() >= 0 ? $this->secondQuarter() : $this->thirdQuarter();
                }
            } catch(Exception $e) {
                return 'error';
            }
        }

        private function firstQuarter() {
            return 'false';
        }

        private function secondQuarter() {
            return (pow($this->x->content(), 2) + pow($this->y->content(), 2)) <= pow($this->r->content(), 2) ? 'true' : 'false';
        }

        private function thirdQuarter() {
            return (abs($this->x->content()) <= $this->r->content()/2) && (abs($this->y->content()) <= $this->r->content()) ? 'true' : 'false';
        }

        private function fourthQuarter() {
            return $this->x->content() - $this->r->content() <= $this->y->content() ? 'true' : 'false';
        }
    }
?>

File: web/WebSecondLab/src/main/java/app/entities/classes/Hit.java
package app.entities.classes;

import app.entities.exceptions.WrongInputException;
import app.entities.interfaces.Containable;
import app.entities.interfaces.Resultable;

public class Hit implements Resultable {
    private final Containable<Float> x;
    private final Containable<Float> y;
    private final Containable<Float> r;

    public Hit(Containable<Float> x, Containable<Float> y, Containable<Float> r) {
        this.x = x;
        this.y = y;
        this.r = r;
    }

    @Override
    public Result result() {
        boolean result;

        try {
            if (x.content() >= 0) {
                result = y.content() > 0 ? firstQuarter() : fourthQuarter();
            } else {
                result = y.content() > 0 ? secondQuarter() : thirdQuarter();
            }
        } catch (WrongInputException e) {
            return Result.ERROR;
        }

        return result ? Result.TRUE : Result.FALSE;
    }

    private boolean firstQuarter() throws WrongInputException {
        return Math.sqrt(Math.pow(x.content(), 2) + Math.pow(y.content(), 2)) <= r.content() / 2;
    }

    private boolean secondQuarter() {
        return false;
    }

    private boolean thirdQuarter() throws WrongInputException {
        return y.content() >= (-0.5 * x.content() - r.content() / 2);
    }

    private boolean fourthQuarter() throws WrongInputException {
        return (Math.abs(x.content()) <= r.content() && Math.abs(y.content()) <= r.content());
    }
}
